presidio-analyzer/presidio_analyzer/recognizer_registry/recognizer_registry.py

```diff
@@ -206,3 +209,52 @@ def remove_recognizer(self, recognizer_name: str) -> None:
             recognizer_name,
         )
         self.recognizers = new_recognizers
+
+    def add_pattern_recognizer_from_dict(self, recognizer_dict: Dict):
+        """
+        Load a pattern recognizer from a Dict into the recognizer registry.
+
+        :param recognizer_dict: Dict holding a serialization of an PatternRecognizer
+
+        :example:
+        >>> registry = RecognizerRegistry()
+        >>> recognizer = { "name": "Titles Recognizer", "supported_language": "de","supported_entity": "TITLE", "deny_list": ["Mr.","Mrs."]} # noqa: E501
+        >>> registry.add_pattern_recognizer_from_dict(recognizer)
+        """
+
+        recognizer = PatternRecognizer.from_dict(recognizer_dict)
+        self.add_recognizer(recognizer)
+
+    def add_recognizers_from_yaml(self, yml_path: Union[str, Path]):
+        r"""
+        Read YAML file and load recognizers into the recognizer registry.
+
+        Example YAML file:
+        ---
+        recognizers:
+          - name: Zip code Recognizer
+            supported_language: de
+            patterns:
+              - name: zip code (weak)
+                regex: '(\b\d{5}(?:\-\d{4})?\b)\'
+                score: 0.01
+            context:
+              - zip
+              - code
+            supported_entity: ZIP
+
+
+        :param yml_path: Path to YAML file
+
+        :example:
+        >>> yaml_file = "recognizers.yaml"
+        >>> registry = RecognizerRegistry()
+        >>> registry.add_recognizers_from_yaml(yaml_file)
+
+        """
+
+        with open(yml_path, "r") as stream:
+            yaml_recognizers = yaml.safe_load(stream)
+
+        for yaml_recognizer in yaml_recognizers["recognizers"]:
```

Comment:
Consider checking if yaml_recognizers is None

Reply:
Surrounded with try except to handle these cases.